vignettes/convert.qmd

```diff
@@ -0,0 +1,346 @@
+```
+
+The package will create this directory if it does not exist on the first run of any function that downloads the data.
+
+<details><summary>Setting data directory for advanced users</summary>
+
+To permanently set the directory for all projects, you can specify the data directory globally by setting the `SPANISH_OD_DATA_DIR` environment variable, e.g. with the following command:
+
+```{r}
+#| eval: false
+usethis::edit_r_environ()
+# Then set the data directory globally, by typing this line in the file:
+```
+
+```
+SPANISH_OD_DATA_DIR = "~/spanish_od_data"
+```
+
+You can also set the data directory locally, just for the current project. Set the 'envar' in the working directory by editing `.Renviron` file in the root of the project:
+
+```{r}
+#| eval: false
+file.edit(".Renviron")
+```
+
+
+</details>
+
+# Introduction {#convert-data}
+
+`spod_get()` returns objects that are only appropriate for small datasets representing a few days of the national OD datasets. We recommend converting the data into analysis-ready formats (`DuckDB` or `Parquet`), allowing you to work with large datasets on a consumer laptop (with 8-16 GB of memory).
+
+**Converting the data to either `DuckDB` or `Parquet` can lead to ~10x speed-up compared with using data frames returned by `spod_get()`, which reads CSV files every time you call it.**
+
+`DuckDB` and `Parquet` are systems for efficiently processing larger-than-memory datasets. For an introduction to them, we recommend materials by Danielle Navarro, Jonathan Keane, and Stephanie Hazlitt: [website](https://arrow-user2022.netlify.app/){target="_blank"}, [slides](https://arrow-user2022.netlify.app/slides){target="_blank"}, and [the video tutorial](https://www.youtube.com/watch?v=YZMuFavEgA4){target="_blank"}.
+
+Learning to use `DuckDB` and `Parquet` is easy for anyone who have ever worked with `{dplyr}` functions such as `select()`, `filter()`, `mutate()`, `group_by()`, `summarise()`, etc. However, since there is some learning curve to master these new tools, we provide some helper functions for novices to get started and easily open the datasets from `DuckDB` and `Parquet`. Please read on. We first show how to convert the data, and then how to use it.
+
+The choice between converting to `DuckDB` and `Parquet` should be made based on your needs:
+
+-   If you plan to download large amounts of data, we recommend `DuckDB`, as it is one big file and it is easier to update it completely if for example you were to first use only dates for 2020, and then later decided to add more dates from 2021. In this case it would be better to delete the database and create it from scratch.
+
+-   If you only want certain dates, do analysis on them and then add more dates later, `Parquet` may be better, as each day is saved in a separate file, just like the original CSV files. Therefore updating a folder of `Parquet` files is as easy as just creating a new file only for the missing date.
```

Comment:
Perhaps rephrase: If you only want certain dates, analyse them and add additional dates later.